### PR TITLE
Enhance Windows Integration: Incorporate Descriptive Security Event Information and Improve KVDB Naming

### PR DESCRIPTION
**Pull request by juliancnn**
|Related issue|
|---|
||

**Description:**

This Pull Request aims to significantly improve the Windows integration of Wazuh-Engine by incorporating more descriptive security event information based on the `event.id`. The additional descriptions are sourced from the KVDB, which has been populated with information available in the Microsoft documentation table, accessible [here](https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/plan/appendix-l--events-to-monitor). The table can also be downloaded from the [download center](https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/plan/appendix-l--events-to-monitor).

**Key Enhancements:**

1. **Descriptive Security Event Information:** For each security event, a detailed description is now available, providing more context and clarity on the nature of the event. This is particularly beneficial for users who need to understand the implications of each event quickly.

2. **Improved KVDB Naming:** The names of the KVDB have been refined to be more explicit and informative. This change indicates when they are used as a table to transform codes into values or retrieve descriptions, enhancing readability and understanding.

3. **Additional Login Failure Information:** For events '4625' and '4776', which represent login failures, the description now includes detailed information on why the authentication failed. This enhancement is crucial for administrators and security personnel to quickly identify and address authentication issues.

**Impact:**

These improvements aim to provide users with more detailed and clear information on security events, aiding in quicker response times and more informed decision-making. The enhancements also contribute to a more user-friendly experience, especially for those who need to interpret and act upon security event data.

**Testing:**

- [x] Verify that each security event now has a detailed description based on its `event.id`.
- [x] Confirm the accuracy and relevance of the added descriptions by cross-referencing with the Microsoft documentation.
- [x] Check the improved naming conventions of the KVDB for clarity and explicitness.
- [x] Validate that events '4625' and '4776' correctly display the reason for authentication failure.

### CODE REVIEW

File: src/engine/ruleset/integrations/windows/decoders/windows-security.yml [hunk 1]
```diff
@@ -27,37 +27,39 @@ normalize:
       - wazuh.decoders: array_append(windows-security)
```
Comment (soynof) [inline src/engine/ruleset/integrations/windows/decoders/windows-security.yml:27]: Please update the metadata compatibility.

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/windows/kvdbs/windows_eventid_to_message.json [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    "6416": "A new external device was recognized by the System",
+    "6417": "The FIPS mode crypto selftests succeeded.",
+    "6418": "The FIPS mode crypto selftests failed.",
+    "6419": "A request was made to disable a device",
+    "6420": "A device was disabled.",
+    "6421": "A request was made to enable a device.",
+    "6422": "A device was enabled.",
+    "6423": "The installation of this device is forbidden by system policy",
+    "6424": "The installation of this device was allowed, after having previously been forbidden by policy."
+}
```
Comment (JcabreraC) [inline src/engine/ruleset/integrations/windows/kvdbs/windows_eventid_to_message.json:409]: Add a line break at the end of the file.

**Review by JcabreraC — CHANGES_REQUESTED** [review]

**Review by JavierBejMen — APPROVED** [review]
LGTM

File: src/engine/ruleset/integrations/windows/kvdbs/windows_eventid_to_message.json (hunk 2 above)
Comment (JavierBejMen) [reply src/engine/ruleset/integrations/windows/kvdbs/windows_eventid_to_message.json]: Done

File: src/engine/ruleset/integrations/windows/decoders/windows-security.yml (hunk 1 above)
Comment (JavierBejMen) [reply src/engine/ruleset/integrations/windows/decoders/windows-security.yml:27]: Done

File: src/engine/ruleset/integrations/windows/decoders/windows-security.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ metadata:
   title: Decoder for Windows security events.
   description: This Windows decoder normalizes Windows security events.
   versions: [Vista, "7", "8", "10", "11", Server 2012, Server 2016, Server 2019, Server 2022]
-  compatibility: This decoder has been tested on Wazuh version 4.3.
+  compatibility: This decoder has been tested on Wazuh version 4.5.
```
[the remark this answers is not in the document]
Comment (juliancnn) [reply src/engine/ruleset/integrations/windows/decoders/windows-security.yml]: done

**Review by NahuFigueroa97 — APPROVED** [review]

File: src/engine/ruleset/integrations/windows/decoders/windows-event.yml [hunk 4]
```diff
@@ -28,6 +28,8 @@ normalize:
 
       - host.name: $windows.System.Computer.#text
 
+      - message: kvdb_get(windows_id_to_message, $event.id)
+
       - _win_log_level:
           "0": "Audit"
```
Comment (soynof) [inline src/engine/ruleset/integrations/windows/decoders/windows-event.yml:34]: Should double quotes be replaced with single quotes, right? And removed double quotes in values.

```
- _win_log_level:
          '0': Audit
          '1': Critical
```

**Review by soynof — CHANGES_REQUESTED** [review]

File: src/engine/ruleset/integrations/windows/decoders/windows-event.yml (hunk 4 above)
Comment (juliancnn) [reply src/engine/ruleset/integrations/windows/decoders/windows-event.yml]: done

**Review by soynof — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**